python/mxnet/gluon/trainer.py

```diff
@@ -187,6 +187,11 @@ def _init_kvstore(self):
             arg_arrays = {param.name: param.data(self._contexts[0]) for param in self._params}
             kvstore, update_on_kvstore = _create_kvstore(config['kvstore'], len(self._contexts),
                                                          arg_arrays)
+            if kvstore and 'async' in kvstore.type and config['update_on_kvstore'] is not None\
```

Comment:
If we are forcing the user to set this param, why don't we set it inside the function itself as default value?

Reply:
If the user does not set that variable explicitly (default way), then I set it to the right value. If the user explicitly sets it to false, then raised the error.